### ISSUE TITLE
Faceting on terms in file sets

### ISSUE THREAD

**Issue by awead** [projecthydra/sufia#2315]
### Descriptive summary

We can return search results from FileSet objects because we're able to do solr joins, and return the GenericWork for which the matching FileSet was found. However, how do we do this for facets?

For example, we want to add a facet for `file_type` which is a term generated during characterization. The term is indexed into the FileSet, but how do we facet on it so that it returns the work object when a file set facet term is chosen?

### Related work

http://stackoverflow.com/questions/24816289/solr-join-and-faceting-possible

**Comment by atz** [comment]
In my experience, solr joins will kill scalability and our ability to reason about queries.  

The most solr-ish answer here is that if you wanted to facet on file_type for work objects, you index that field in your work documents (in addition to wherever else you want it).

**Comment by cbeer** [comment]
Is that still true with the enhancements they've made in Solr 5 + 6?

**Comment by awead** [comment]
@atz yeah, that was my initial solution to the problem, similar to what we're doing with FileSet > File pairings viz. characterization metadata.

It's an open question at this point, and this issue is here to facilitate discussion.

**Comment by atz** [comment]
It will always be true that indexing what you want ahead of time will outperform runtime computation.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  That is the point of indexing.  Whether join performance in Solr 6 is improved or "better enough" is not something I would pursue.
